lib/src/home_directory.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:path/path.dart' as p;
+
+/// Designated home path for the current user and operating system.
+///
+/// May be `null` on some platforms. Flutter users should consider using the
+/// [path_provider](https://pub.dartlang.org/packages/path_provider) package,
+/// which has support for Android and iOS.
+String get homePath {
+  if (Platform.isMacOS || Platform.isLinux || Platform.isAndroid) {
+    // Source: http://en.wikipedia.org/wiki/Home_directory.
+    return Platform.environment['HOME'];
+  }
+  if (Platform.isWindows) {
+    // Source: https://en.wikipedia.org/wiki/Special_folder.
+    return Platform.environment['UserProfile'];
+  }
+  return null;
+}
+
+/// Returns an application data path for the current user and operating system.
+///
+/// May return `null` on some platforms without a directory for this purpose.
+/// Flutter users should consider using the
+/// [path_provider](https://pub.dartlang.org/packages/path_provider) package,
+/// which has support for Android and iOS.
+String appDataPath(String path) {
+  if (Platform.isWindows) {
+    return p.join(Platform.environment['APPDATA'], path);
+  }
+  if (homePath != null) {
+    return p.join(homePath, '.$path');
```

Comment:
This worries me.

Mac: why not ~/Library/Application support?

Also: adding a `dot` before a path String feels...weird...

Reply:
re: OSX, you might not have privileges to write there, you usually do to APPDATA.

I'm definitely open to other suggestions though!

Reply:
I have zero domain knowledge here, but all the links I found online suggest some path under `~/Library`, and most of them specifically call out `~/Library/Application Support`.